lib/streamlit/runtime/redis_session_storage.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import os
+import pickle
+
+from pyredis import RedisConnection
+
+from streamlit.runtime.session_manager import SessionInfo, SessionStorage
+
+
+class RedisSessionStorage(SessionStorage):
+    """A SessionStorage that stores sessions in redis.
+    """
+    def __init__(
+        self,
+    ) -> None:
+        """Instantiate a new RedisSessionStorage.
+        """
+        redis_url = os.getenv("REDIS_URL")  # host:port/db
+        host, port_db = redis_url.replace("redis://", "").split(":")
+        port, db = port_db.split("/")
+        self._cache = RedisConnection(host=host, port=port, db=db)
```

Comment:
I guess you don't have to parse the host / port.
You can use [from_url](https://redis-py.readthedocs.io/en/stable/connections.html#redis.Redis.from_url).